test/load-generator/boulder-calls.go

```diff
@@ -582,3 +584,75 @@ func postAsGet(s *State, ctx *context, url string, latencyTag string) (*http.Res
 
 	return s.post(url, requestPayload, ctx.ns, latencyTag, http.StatusOK)
 }
+
+func popCertificate(ctx *context) string {
+	certIndex := mrand.Intn(len(ctx.certs))
+	certURL := ctx.certs[certIndex]
+	ctx.certs = append(ctx.certs[:certIndex], ctx.certs[certIndex+1:]...)
+	return certURL
+}
+
+func getCert(s *State, ctx *context, url string) ([]byte, error) {
+	latencyTag := "/acme/cert/{serial}"
+	resp, err := postAsGet(s, ctx, url, latencyTag)
+	if err != nil {
+		return nil, fmt.Errorf("%s bad response: %s", url, err)
+	}
+	defer resp.Body.Close()
+	return ioutil.ReadAll(resp.Body)
+}
+
+// revokeCertificate removes a certificate url from the context, retrieves it,
+// and sends a revocation request for the certificate to the ACME server.
+// The revocation request is signed with the account key rather than the certificate
+// key.
+func revokeCertificate(s *State, ctx *context) error {
+	if len(ctx.certs) < 1 {
+		return errors.New("No certificates in the context that can be revoked")
+	}
+
+	certURL := popCertificate(ctx)
+	certPEM, err := getCert(s, ctx, certURL)
+	if err != nil {
+		return err
+	}
+
+	pemBlock, _ := pem.Decode(certPEM)
+	revokeObj := struct {
+		Certificate string
+		Reason      int
+	}{
+		Certificate: base64.URLEncoding.EncodeToString(pemBlock.Bytes),
+		Reason:      0, // randomly pick one probably
+	}
+
+	revokeJSON, err := json.Marshal(revokeObj)
+	if err != nil {
+		return err
+	}
+	revokeURL := s.directory.EndpointURL(acme.RevokeCertEndpoint)
+	jws, err := ctx.signKeyIDV2Request(revokeJSON, revokeURL) // it'd be nice to also use the cert key randomly?
+	if err != nil {
+		return err
+	}
+	requestPayload := []byte(jws.FullSerialize())
+
+	resp, err := s.post(
+		revokeURL,
+		requestPayload,
+		ctx.ns,
+		"/acme/revoke-cert",
```

Comment:
Can you use `revokeURL` here instead?
```suggestion
		revokeURL,
```

Reply:
`revokeURL` contains the full URL which is presumably not what we want here for the latency tag. Looking through the existing code there seems to be a split between using the `acme.XXXEndpoint` vars and using `/acme/XXX` strings. We should probably decide on using one format for the latency tag and stick with it.

Reply:
I think I preferred the full URL except in the case where I knew there was a parameter in the URL that changed between requests.

In this case since there isn't a param the full URL seems OK? I'm not super attached one way or the other.